src/DataStructures/DataBox/Actions/SetData.hpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+
+#include "Parallel/Printf.hpp"
+
+namespace db {
+namespace Actions {
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Mutate the DataBox tags in `TagsList` according to the `data`.
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - All tags in `TagsList`
+ */
+template <typename TagsList>
+struct SetData {
+  template <typename... DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent,
+            Requires<sizeof...(DbTags) != 0> = nullptr>
```

Comment:
Please explain, just here on GitHub so I can learn, how this TMP Requires<> = nullptr works?

Reply:
@geoffrey4444  A template with parameter `Requires<CONDITION> = nullptr` will only compile if `CONDITION` evaluates to `true`, and will produce a compiler error if `CONDITION` is `false`. 
It works by transforming `Requires<CONDITION>` to the type `std::nullptr_t` if the CONDITION is `true` and to a bogus type if CONDITION is `false`.   The `= nullptr` is a default template parameter so that you don't need to specify a parameter when using the template.    See the documentation of `Utilities/Requires.hpp` in the `Utilities` topic for more details

Reply:
@kidder asked me to "fact check" this in person. He's very close. A `false` condition removes a function from overload resolution so the compiler is not able to use that function. What this means is you can disable a function from being used based on a certain condition, and have a separate one be used if the condition is false. You get a compiler error if *all* functions with that name were not usable because of type mismatches or `Requires` failures. The compiler error actually just needs to be `no such function found`, but since that would make it nearly impossible to figure out what is happening the shepherds of dragons (compiler devs) have taken mercy on us and provided lists of functions with matching names and why they couldn't be used.